cosmwasm/contracts/pyth/src/contract.rs

```diff
@@ -115,6 +134,66 @@ fn submit_vaa(
     process_batch_attestation(deps, env, &batch_attestation)
 }
 
+fn execute_governance_instruction(
+    mut deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    data: &Binary,
+) -> StdResult<Response> {
+    let vaa = parse_vaa(deps.branch(), env.block.time.seconds(), data)?;
+
+    execute_governance_instruction_from_vaa(deps, env, info, &vaa)
+}
+
+/// Helper function to improve testability of governance instructions (so we can unit test without wormhole).
+fn execute_governance_instruction_from_vaa(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    vaa: &ParsedVAA,
+) -> StdResult<Response> {
+    let state = config_read(deps.storage).load()?;
+
+    // store updates to the config as a result of this action in here.
+    let mut updated_config: ConfigInfo = state.clone();
+    verify_vaa_from_governance_source(&state, vaa)?;
+
+    if vaa.sequence <= state.governance_sequence_number {
+        return Err(PythContractError::OldGovernanceMessage)?;
+    } else {
+        updated_config.governance_sequence_number = vaa.sequence;
+    }
+
+    let data = &vaa.payload;
+    let instruction = GovernanceInstruction::deserialize(&data[..])
+        .map_err(|_| PythContractError::InvalidGovernancePayload)?;
+
+    if instruction.target_chain_id != state.chain_id && instruction.target_chain_id != 0 {
+        return Err(PythContractError::InvalidGovernancePayload)?;
+    }
+
+    let response = match instruction.action {
+        SetFee { val, expo } => {
+            updated_config.fee = (val as u128)
+                .checked_mul(
+                    10_u128
+                        .checked_pow(expo as u32)
```

Comment:
Can we do a checked conversion here instead of `expo as u32`? `as` will truncate the value.